pysat/_instrument.py

```diff
@@ -578,8 +578,8 @@ def __setitem__(self, key, new):
                 indict = {}
                 for i, dim in enumerate(self[key[-1]].dims):
                     indict[dim] = key[i]
-                    # if dim == 'Epoch':
-                    #     indict[dim] = self.index[key[i]]
+                    if dim == epoch_name:
+                        indict[dim] = self.index[key[i]]
```

Comment:
Turns out restoring this bit of code breaks three of the unit tests.
- test_instrument.py::TestBasicsXarray::test_setting_partial_data_by_index_and_name FAILED
- test_instrument.py::TestBasicsXarray::test_setting_partial_data_by_datetime_and_name FAILED
- test_instrument.py::TestBasicsXarray::test_setting_partial_data_by_datetime_slicing_and_name FAILED

Removing and finding another solution

Reply:
I looked back at then if dim == 'Epoch' was first modified recently and it was commented before that change. You are correct to remove it. Thanks Jeff.